intro-js.md

```diff
@@ -0,0 +1,149 @@
+# Introduction
+Welcome to the weird and wonderful world of JavaScript development. 
+JavaScript, as a language, is used primarily in the development of web applications and websites. 
+
+## Why is JavaScript weird? 
+
+1. Lack of types
+
+JavaScript is a *loosely typed* or *dynamically typed* language
+
+```javascript
+var a = 42; //a is a number here
+var a = 'random' //a is a string here
+var a = true //a is a boolean here
+```
+
+This makes JavaScript development cumbersome and bug ridden, but there are improvements being made. 
+
+A solution to this has been provided through [TypeScript](https://www.typescriptlang.org/docs/home.html)
+
+Angular 2 utilizes TypeScript extensively.
+
+2. Lack of classes
+JavaScript did not have the concept of classes when it was first written.
+
+Classes have now been provided as part of [ECMAScript 2015](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes), but this is just syntactic sugar. 
+
+Example class
+```javascript 
+class Shape {
+    constructor(param1, param2) {
```

Comment:
Not entirely sure, but maybe because we're assuming a technical background  of some programming language for the students? Would we need to explain how functions and objects work in that case?